ui/apps/dev-server-ui/src/components/Timeline/historyParser/historyParser.ts

```diff
@@ -31,7 +31,7 @@ export class HistoryParser {
 
     let node: HistoryNode;
     if (this.history[rawItem.groupID]) {
-      node = { ...this.history[rawItem.groupID] };
+      node = { ...(this.history[rawItem.groupID] as HistoryNode) };
```

Comment:
This will type narrow, avoiding the need for casting:

```ts
    let node: HistoryNode;
    const existingNode = this.history[rawItem.groupID];
    if (existingNode) {
      node = { ...existingNode };
    } else {
```